src/main/java/in/projecteka/consentmanager/user/UserService.java

```diff
@@ -124,18 +114,27 @@
     public Mono<Session> update(UpdateUserRequest updateUserRequest, String sessionId) {
         return signupService.getUserName(sessionId)
                 .switchIfEmpty(Mono.error(new InvalidRequestException("user not verified")))
-                .flatMap(userName -> updatedSessionFor(updateUserRequest, userName));
+                .flatMap(userName -> updatedSessionFor(updateUserRequest.getPassword(), userName));
     }
 
-    private Mono<Session> updatedSessionFor(UpdateUserRequest updateUserRequest, String userName) {
+
+    public Mono<Session> updatePasswordFor(String password, String userName) {
+        return getSession(password, userName, failedToUpdatePassword());
+    }
+
+    private Mono<Session> updatedSessionFor(String password, String userName) {
+        return getSession(password, userName, failedToUpdateUser());
```

Comment:
why send the error as parameter, it is already a static import...use it directly where it is needed

Reply:
Updated